Support for URL/web link drag and drop

WizardCM opened the pull request:
### Description
Extending the ability to drag files into the OBS window to automatically add it to the scene, this adds the ability to do the same for web links (especially when dragged from a web browser).

Web links can be built into any web element, like a button or thumbnail. The web developer can also define the width, height, and source name for when the source is added. See the link below when testing. Dragged links default to canvas size.

![](http://scr.wzd.li/scr/2019-12-28_19-03-09.gif)

### Motivation and Context
Users should be able to efficiently build their overlays. With web overlays becoming more and more popular, this PR aims to improve the workflow of importing them.

### How Has This Been Tested?
Drag any one of these into OBS: https://wizardcm.github.io/OBS-Drag-Demo/

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.

Lange commented:
My gut feeling is that all the URL query parameters should have something like an `obs-` prefix, otherwise there will be too many false positives.

WizardCM commented:
I have updated the PR to have a confirmation dialog (Yes/No) and to require an `obs-` prefix for width, height, and name parameters. The [demo page](https://wizardcm.github.io/OBS-Drag-Demo/) has been updated to reflect this.

JohannMG commented:
This is GREAT. 

I like the prefix but perhaps we can make the `obs-` into something less OBS-specific so other broadcast software and more vendors can support this? 😃

WizardCM commented:
> This is GREAT.
> 
> I like the prefix but perhaps we can make the `obs-` into something less OBS-specific so other broadcast software and more vendors can support this? 😃

Hmm, not a bad idea. `overlay-` maybe? I considered `live-` but that feels limiting. Could even be `layer-`?

JohannMG commented:
Something I'm drafting right now would use `bss-` for *broadcast stream setup* but here I think `overlay-` and `layer-` both sound very correct

randallb commented:
Just a heads up— I work with Facebook, and our project is really excited to use this feature. Thanks for the hard work! 👍

WizardCM commented:
Alrighty, per the request, I've changed the prefix to `layer-`. [Demo page](https://wizardcm.github.io/OBS-Drag-Demo/) updated again.

I also added a basic Escape = No and Enter = Cancel to the dialog, **however** I noticed that because of how drag/drop works in Windows, the dialog doesn't automatically get focused (even though Qt documentation says it should). Ideally I'd like to do this without making the dialog async, but as .setFocus() didn't fix the issue, I worry I might have to. I someone else could confirm this, that'd be great.

jp9000 commented:
I cleaned things up a bit.  To raise the main window you use activateWindow().

WizardCM commented:
Aha, activateWindow() is exactly what I was looking for, thanks for the info, and appreciate the cleanup.
